File: Cores/OSOL.Extremum.Core.DotNet/Random/Distributions/Normal.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace OSOL.Extremum.Core.DotNet.Random.Distributions
{
    public interface INormal
    {
        double GetNormal(double mu, double sigma);   
    }

    public static class NormalFunctions
    {
        public static Dictionary<string, double> Diameter(this INormal GoRN, Dictionary<string, Tuple<double, double>> area) =>
            area.ToDictionary(kvp => kvp.Key,
                kvp => GoRN.GetNormal(kvp.Value.Item1, kvp.Value.Item2));
    }

}

File: Cores/JVM/src/main/scala/OSOL/Extremum/Cores/JVM/Random/Distributions/Normal.scala
package OSOL.Extremum.Cores.JVM.Random.Distributions

trait Normal {

  def getNormal(mu: java.lang.Double, sigma: java.lang.Double): java.lang.Double

  final def getNormal(area: Map[String, (java.lang.Double, java.lang.Double)]): Map[String, java.lang.Double] =
    area.map { case (key, (mu, sigma)) => (key, getNormal(mu, sigma)) }

  def getNormalScala(mu: Double, sigma: Double): Double =
    getNormal(mu, sigma)

  final def getNormalScala(area: Map[String, (Double, Double)]): Map[String, Double] =
    area.map { case (key, (mu, sigma)) => (key, getNormalScala(mu, sigma)) }

}